File: CS/BlazorGridUnboundColumns/Data/WeatherForecastService.cs
namespace BlazorGridUnboundColumns.Data {
    public class WeatherForecastService {
        public Task<WeatherForecast[]> GetForecastAsync(DateTime startDate) {
            var rng = new Random();
            return Task.FromResult(Enumerable.Range(1, 5).Select(index => new WeatherForecast {
                Date = startDate.AddDays(index),
                TemperatureC = rng.Next(-20, 30),
            }).ToArray());
        }
    }
}

File: VB/BlazorGridUnboundColumns/Data/WeatherForecastService.vb
Namespace BlazorGridUnboundColumns.Data

    Public Class WeatherForecastService

        Public Function GetForecastAsync(ByVal startDate As DateTime) As Task(Of WeatherForecast())
            Dim rng = New Random()
            Return Task.FromResult(Enumerable.Range(1, 5).[Select](Function(index) New WeatherForecast With {.[Date] = startDate.AddDays(index), .TemperatureC = rng.[Next](-20, 30)}).ToArray())
        End Function
    End Class
End Namespace
